### PR TITLE
new extension build process

### PR DESCRIPTION
**Pull request by julian1**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set -x
```
Comment (danfruehauf) [inline build.sh]: You probably want the first line to be `#!/bin/bash`, then if you want `-x`, change it to `#!/bin/bash -x`

File: pom.xml [hunk 2]
```diff
@@ -1,90 +1,56 @@
...
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
     <modelVersion>4.0.0</modelVersion>
     <groupId>au.org.emii</groupId>
     <artifactId>geoserver</artifactId>
-    <packaging>war</packaging>
-    <version>2.6.2</version>
+    <packaging>pom</packaging>
+    <version>1.0.0</version>
```
Comment (danfruehauf) [inline pom.xml]: You know what I think about the choice of version here.

I personally think it should reflect somehow the version of Geoserver that's encapsulated in the build.

File: build.sh (hunk 1 above)
Comment (julian1) [inline build.sh]: fixed!

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash -x
+git submodule update --init
+mvn clean install -U -Dmaven.test.skip=true
```
Comment (jonescc) [inline build.sh]: This builds and installs to the local repository - did you want to do that?

File: pom.xml [hunk 4]
```diff
@@ -1,90 +1,56 @@
...
+        </dependencies>
+    </dependencyManagement>
+
+     <!-- gs-xslt plugin -->
+    <repositories>
         <repository>
-            <id>${internal.repository.id}</id>
-            <url>${internal.repository.url}</url>
+            <id>opengeo</id>
+            <url>http://repo.boundlessgeo.com/main</url>
```
Comment (jonescc) [inline pom.xml]: So we've decided not to bother with Archiva for caching of artifacts?

**Comment by danfruehauf** [comment]
@jonescc Please provide more code related feedback, I'll do some end to end testing of that with no_db, see how we go.

**Comment by danfruehauf** [comment]
Alrighty, fine to be merged by me. @jonescc Any more comments?

**Comment by jonescc** [comment]
No, apart from comments made, looks pretty good to me.  Nice work @julian1

**Comment by danfruehauf** [comment]
So I'll just let @julian1 address the comments you've made.

File: build.sh (hunk 3 above)
Comment (julian1) [inline build.sh]: I almost removed this script file, as it's not used for Jenkins which is the correct place to build. So at the moment it's serves as a documented quickstart, to enable another dev picking this up to know what the required git submodule incantations are.

File: pom.xml (hunk 4 above)
Comment (julian1) [inline pom.xml]: I would like to see this for maven projects. But feel this is probably best addresssed in a retro.

File: build.sh (hunk 3 above)
Comment (julian1) [inline build.sh]: @jonescc On seconds thoughts I might put this into the README

Comment (julian1) [inline build.sh]: @jonescc updated.

**Comment by danfruehauf** [comment]
I think it's all addressed. Can I merge?

**Pull request merged**